src/sentry/static/sentry/app/stores/guideStore.tsx

```diff
@@ -90,68 +97,80 @@ const GuideStore = Reflux.createStore({
     this.trigger(this.state);
   },
 
-  onRegisterAnchor(anchor) {
-    this.state.anchors.add(anchor);
+  onRegisterAnchor(target: string) {
+    this.state.anchors.add(target);
     this.updateCurrentGuide();
   },
 
-  onUnregisterAnchor(anchor) {
-    this.state.anchors.delete(anchor);
+  onUnregisterAnchor(target: string) {
+    this.state.anchors.delete(target);
     this.updateCurrentGuide();
   },
 
-  recordCue(id) {
+  recordCue(id: number) {
     const data = {
       eventKey: 'assistant.guide_cued',
       eventName: 'Assistant Guide Cued',
       guide: id,
+      organization_id: this.state.orgId,
     };
-    if (this.state.org) {
-      data.organization_id = this.state.org.id;
-    }
     trackAnalyticsEvent(data);
   },
 
-  updatePrevGuide(bestGuide) {
+  updateGuidesWithContent(data: GuideData) {
+    try {
+      const content = getGuideContent();
+      const guides = Object.keys(data).reduce((acc, key) => {
+        if (key in content) {
+          acc[key] = {...data[key], ...content[key]};
+        }
+        return acc;
+      }, {});
+
+      this.state.guides = guides;
+      this.updateCurrentGuide();
+    } catch (e) {
+      Sentry.captureException(e);
+    }
+  },
+
+  updatePrevGuide(bestGuide: Guide) {
+    const {prevGuide} = this.state;
     if (!bestGuide) {
       return;
     }
 
-    if (!this.state.prevGuide || this.state.prevGuide.id !== bestGuide.id) {
+    if (!prevGuide || prevGuide.id !== bestGuide.id) {
       this.recordCue(bestGuide.id);
       this.state.prevGuide = bestGuide;
     }
   },
 
+  /**
+   * Logic to determine if a guide is shown:
+   *
+   *  - If any required target is missing, don't show the guide
+   *  - If the URL ends with #assistant, show the guide
+   *  - If the user has already seen the guide, don't show the guide
+   *  - Otherwise show the guide
+   */
   updateCurrentGuide() {
-    // Logic to determine if a guide is shown:
-    // 1. If any required target is missing, don't show the guide.
-    // 2. If the URL ends with #assistant, show the guide.
-    // 3. If the user has seen the guide or is an old non-superuser, don't show the guide.
-    // 4. Otherwise show the guide.
-
-    const availableTargets = [...this.state.anchors].map(a => a.props.target);
+    const {anchors, guides, forceShow} = this.state;
 
-    // sort() so that we pick a guide deterministically every time this function is called.
-    let guideKeys = Object.keys(this.state.guides)
+    let availableGuides = Object.keys(guides)
       .sort()
-      .filter(key => {
-        return this.state.guides[key].required_targets.every(
-          t => availableTargets.indexOf(t) >= 0
-        );
-      });
+      .filter(key =>
+        guides[key].required_targets.every((target: string) => anchors.has(target))
```

Comment:
the `guides` state must not be typed right here if you're having to type `target` to string, it should be aware of the type of `required_targets`

Reply:
yeah, do I need to pull `state` out of the `init` in order for it to be typed?